### PR TITLE
Fiber nulling module

### PR DESCRIPTION
**Pull request by yinzi-xin**
This adds a fiber nulling module with generic/custom fiber nuller, vortex fiber nuller, and photonic lantern nuller classes.

### CODE REVIEW

**Comment by ehpor** [comment]
Discussed this offline yesterday with @yinzi-xin. We discussed issues and fixes for those issues.

(Just making a note of this for the sake of documentation.)

**Comment by yinzi-xin** [comment]
I think I'm done making edits to the code. Can you run the tests again that identify where all the issues (e.g. missing whitespaces) are?

**Comment by ehpor** [comment]
@yinzi-xin They are running now.

**Comment by ehpor** [comment]
Test failures are unrelated to this PR and will be fixed by #245 and #246.

That being said: there should be some tests for the new code that you're adding @yinzi-xin. I'll have to think what constitutes a correct test. Probably injecting one of the modes of the lantern and confirming that that mode is completely injected into one of the output waveguides. And for the vortex fiber nuller: testing the fact that it suppresses all light (within a tolerance) on a symmetric pupil (say, circular).

File: hcipy/optics/photonic_lantern.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    '''A generic photonic lantern
+
+    Parameters
+    ----------
+    lantern_modes : ModeBasis
+        The modes corresponding to the lantern ports.
+    wavelength : scalar
+        The wavelength of the simulation.
+    '''
+    def __init__(self, lantern_modes, normalize_modes=True):
```
Comment (ehpor) [inline hcipy/optics/photonic_lantern.py:17]: `normalize_modes`? It's not used by the function, and it's not in the docstring.

**Review by ehpor — CHANGES_REQUESTED** [review]
Only one comment remaining. Otherwise this is perfect.

I made some formatting changes, so you'll have to `git pull` before you make more changes. @yinzi-xin

**Comment by ehpor** [comment]
@yinzi-xin Thanks for your first contribution! 🎉

**Pull request merged**